moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/src/simple_sampler.cpp

```diff
@@ -92,34 +83,25 @@ SimpleSampler::getLocalTrajectory(const moveit::core::RobotState& current_state,
   // Delete previous local trajectory
   local_trajectory.clear();
 
-  // Determine current local trajectory based on configured behavior
-  if (pass_through_)
-  {
-    // Use reference_trajectory as local trajectory
-    local_trajectory.append(*reference_trajectory_, 0.0, 0, reference_trajectory_->getWayPointCount());
-  }
-  else
+  // Get next desired robot state
+  moveit::core::RobotState next_desired_goal_state = reference_trajectory_->getWayPoint(next_waypoint_index_);
+
+  // Check if state is reached
+  if (next_desired_goal_state.distance(current_state, joint_group_) <= WAYPOINT_RADIAN_TOLERANCE)
   {
-    // Get next desired robot state
-    moveit::core::RobotState next_desired_goal_state = reference_trajectory_->getWayPoint(next_waypoint_index_);
-
-    // Check if state is reached
-    if (next_desired_goal_state.distance(current_state, joint_group_) <= WAYPOINT_RADIAN_TOLERANCE)
-    {
-      // Update index (and thus desired robot state)
-      next_waypoint_index_ = std::min(next_waypoint_index_ + 1, reference_trajectory_->getWayPointCount() - 1);
-    }
-
-    // Construct local trajectory containing the next global trajectory waypoint
-    local_trajectory.addSuffixWayPoint(reference_trajectory_->getWayPoint(next_waypoint_index_),
-                                       reference_trajectory_->getWayPointDurationFromPrevious(next_waypoint_index_));
+    // Update index (and thus desired robot state)
+    next_waypoint_index_ = std::min(next_waypoint_index_ + 1, reference_trajectory_->getWayPointCount() - 1);
```

Comment:
Is there a chance this index will end up being < 0 if getWayPointCount returned 0?

Reply:
I added handling for this, just in case.

I'm sure you'll see that I used a string for the feedback message (type is moveit_msgs::action::LocalPlanner::Feedback). I know an enum would be better but I don't want to wait for a new PR to moveit_msgs at the moment. Will make an issue for it.